src/Microsoft.TestPlatform.CommunicationUtilities/SocketCommunicationManager.cs

```diff
@@ -138,7 +138,11 @@ public async Task AcceptClientAsync()
         /// <returns>True if Client is connected, false otherwise</returns>
         public bool WaitForClientConnection(int clientConnectionTimeout)
         {
-            return this.clientConnectedEvent.WaitOne(clientConnectionTimeout);
+            Stopwatch stopWatch = Stopwatch.StartNew();
+            var result = this.clientConnectedEvent.WaitOne(clientConnectionTimeout);
+            EqtTrace.Verbose("SocketCommunicationManager.WaitForClientConnection took: {0} ms, with {1} ms timeout, and finished with {2}", stopWatch.ElapsedMilliseconds, clientConnectionTimeout, result);
```

Comment:
Nit: We could add a period at the end of the sentence.

```suggestion
            EqtTrace.Verbose("SocketCommunicationManager.WaitForClientConnection took: {0} ms, with {1} ms timeout, and finished with {2}.", stopWatch.ElapsedMilliseconds, clientConnectionTimeout, result);
```